src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultEnumerableConverter.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal class JsonEnumerableT<T> : ICollection<T>, IEnumerable<T>, IList<T>, IReadOnlyCollection<T>, IReadOnlyList<T>
+    {
+        List<T> _list;
+
+        public JsonEnumerableT(IList sourceList)
+        {
+            // TODO: Change sourceList from IList to List<T> so we can do a direct assignment here.
+            _list = new List<T>();
+
+            foreach (object item in sourceList)
+            {
+                if (item is T itemT)
+                {
+                    _list.Add(itemT);
+                }
+            }
+        }
+
+        public T this[int index] { get => (T)_list[index]; set => _list[index] = value; }
+
+        public int Count => _list.Count;
+
+        public bool IsReadOnly => false;
+
+        public void Add(T item)
+        {
+            _list.Add(item);
+        }
+
+        public void Clear()
+        {
+            _list.Clear();
+        }
+
+        public bool Contains(T item)
+        {
+            return _list.Contains(item);
+        }
+
+        public void CopyTo(T[] array, int arrayIndex)
+        {
+            _list.CopyTo(array, arrayIndex);
+        }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            return _list.GetEnumerator();
+        }
+
+        public int IndexOf(T item)
+        {
+            return _list.IndexOf(item);
+        }
+
+        public void Insert(int index, T item)
+        {
+            _list.Insert(index, item);
+        }
+
+        public bool Remove(T item)
+        {
+            return _list.Remove(item);
+        }
+
+        public void RemoveAt(int index)
+        {
+            _list.RemoveAt(index);
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+    }
+
+    internal sealed class DefaultEnumerableConverter : JsonEnumerableConverter
+    {
+        public override IEnumerable CreateFromList(Type elementType, IList sourceList)
+        {
+            Type t = typeof(JsonEnumerableT<>).MakeGenericType(elementType);
```

Comment:
Minor: for performance it is possible to avoid this call to `MakeGenericType` by getting the "property policy" class for this property and having it create the `JsonEnumerableT<TElement>` for you.

You'd have to add a virtual method on the `JsonPropertyInfo` class and then provide an implementation in the derived `JsonPropertyInfoCommon<,,>` which has the element type so you can just return via `return typeof(JsonEnumerableT<TRuntimeProperty>)'.

In any case, you can leave as-is and I can do this with my pending perf work that we discussed.

Reply:
Nice. I'll let you scoop this up in your perf work.